go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -696,12 +696,26 @@ func (bnc *BaseNetworkController) doesNetworkRequireIPAM() bool {
 	return util.DoesNetworkRequireIPAM(bnc.NetInfo)
 }
 
-func (bnc *BaseNetworkController) getPodNADNames(pod *kapi.Pod) []string {
+func (bnc *BaseNetworkController) getPodNADNames(pod *kapi.Pod) ([]string, error) {
 	if !bnc.IsSecondary() {
-		return []string{types.DefaultNetworkName}
+		return []string{types.DefaultNetworkName}, nil
+	}
+	if bnc.IsPrimaryNetwork() {
+		namespacePrimaryNetwork, err := bnc.getActiveNetworkForNamespace(pod.Namespace)
```

Comment:
Why do we need to do this?

Isn't this network controller the one for the active network for the namespace?

In which cases do we expect `namespacePrimaryNetwork` to be different than `bnc.NetInfo`?

Reply:
Comments always help though

Reply:
added comment to explain this.